query_optimizer/logical/Aggregate.cpp

```diff
@@ -72,6 +75,35 @@ std::vector<E::AttributeReferencePtr> Aggregate::getReferencedAttributes() const
   return referenced_attributes;
 }
 
+LogicalPtr Aggregate::copyWithNewInputExpressions(
+      const std::vector<E::ExpressionPtr>& input_expressions) const {
```

Comment:
I like the nicer Zuyu :-) You are right @zuyu -- we should try to use the standard here